### PR TITLE
Atmega common mod periph conf

### PR DESCRIPTION
**Pull request by A-Paul**
This is a preparation for implemeting a PWM driver for the *atmeg328p* MCU. The *periph_conf* has not been prepared for board specific configurations.

I just placed a *periph_conf.h* in the common include directory, which works fine. If it is copied to the board specific include directory then the make system selects that instead.

@miri64 for the board definitions you have placed a empty *board.h* in every specific include directory and none in the common. Are there reasons to prefer that schema?

### CODE REVIEW

**Comment by haukepetersen** [comment]
Why not go all the way here? The common file already contains a lot of `#ifdef' blocks, so why not make them specific to each atmega based board right away? This would be much cleaner...

**Comment by A-Paul** [comment]
> Why not go all the way here? ...

@haukepetersen because that would be a shitload of work for me. Just kidding...
I was already wondering if having those conditionals in "common" might signify it is "uncommon". ;)
I can try to clean this up. In the current state the arduino section is rather confusing to me.  prefer to do further steps in separate PRs based on this.

**Comment by roberthartung** [comment]
What about #7542 ?

File: boards/arduino-atmega-common/include/periph_conf_common.h [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+ * so this is the only one we can use here.
+ *
+ * arduino mega2560:
+ * The timer driver only supports the four 16-bit timers (Timer1, Timer3,
+ * Timer4, Timer5), so those are the only onces we can use here.
+ *
+ * @{
+ */
+#ifdef CPU_ATMEGA328P
+#define TIMER_NUMOF         (2U)
```
Comment (smlng) [inline boards/arduino-atmega-common/include/periph_conf_common.h]: looks like there actually only 1 timer, right?

File: boards/arduino-atmega-common/include/periph_conf_common.h [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+
+#ifdef CPU_ATMEGA2560
+#define TIMER_NUMOF         (2U)
+
+#define TIMER_0             MEGA_TIMER1
+#define TIMER_0_MASK        &TIMSK1
+#define TIMER_0_FLAG        &TIFR1
+#define TIMER_0_ISRA        TIMER1_COMPA_vect
+#define TIMER_0_ISRB        TIMER1_COMPB_vect
+#define TIMER_0_ISRC        TIMER1_COMPC_vect
```
Comment (smlng) [inline boards/arduino-atmega-common/include/periph_conf_common.h]: could be simplified to this line by removing `#ifdef CPU_ATMEGA328P` above

**Comment by miri64** [comment]
From the title I can't tell what this PR is about. Please update.

**Comment by A-Paul** [comment]
Closing because of already restructured *boards/common* tree.

**Pull request closed without merging**